Purchases/Networking/Operations/GetOfferingsOperation.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GetOfferingsOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class GetOfferingsOperation: CacheableNetworkOperation {
+
+    private let offeringsCallbackCache: CallbackCache<OfferingsCallback>
+    private let configuration: AppUserConfiguration
+
+    init(configuration: UserSpecificConfiguration,
+         offeringsCallbackCache: CallbackCache<OfferingsCallback>) {
+        self.configuration = configuration
+        self.offeringsCallbackCache = offeringsCallbackCache
+
+        super.init(configuration: configuration, individualizedCacheKeyPart: configuration.appUserID)
+    }
+
+    override func begin() {
+        self.getOfferings()
+    }
+
+}
+
+private extension GetOfferingsOperation {
+
+    func getOfferings() {
+        guard let appUserID = try? configuration.appUserID.escapedOrError() else {
+            self.offeringsCallbackCache.performOnAllItemsAndRemoveFromCache(withCacheable: self) { callback in
+                callback.completion(nil, ErrorUtils.missingAppUserIDError())
+            }
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/offerings"
+        httpClient.performGETRequest(serially: true,
+                                     path: path,
+                                     headers: authHeaders) { statusCode, maybeResponse, maybeError in
+            if maybeError == nil && statusCode < HTTPStatusCodes.redirect.rawValue {
+                self.offeringsCallbackCache.performOnAllItemsAndRemoveFromCache(withCacheable: self) { callbackObject in
+                    callbackObject.completion(maybeResponse, nil)
+                }
+                return
+            }
+
+            let errorForCallbacks: Error
+            if let error = maybeError {
+                errorForCallbacks = ErrorUtils.networkError(withUnderlyingError: error)
+            } else if statusCode >= HTTPStatusCodes.redirect.rawValue {
```

Comment:
I feel like some of this error handling could be done by `HTTPClient`. Maybe the callback could just be a transformation (`map`) from `RawResponse` -> `ResponseType` (or `RawResponse` -> Result<ResponseType, Error>` (`flatMap`), and it's only called if the request didn't actually fail.
Just a suggestion for a future improvement.